### ISSUE TITLE
Allow full-offline startup after resolved libraries once 

### ISSUE THREAD

**Issue by Ghost-chu** [AlessioDP/libby#41]
In some cases, the server needs to be started without an (stable) internet connection. SpigotLibraryLoader is able to do this, but libby requires an robust internet connection every time it is started in order to read the dependency files (even if cached files already exist).

**Comment by frengor** [comment]
Are you using the transitive dependency resolution feature?

**Comment by Ghost-chu** [comment]
Yes, I'm using&nbsp;transitive dependency resolution
&nbsp;


Ghost_chu
***@***.***



&nbsp;




------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;; 
Date: 2024年3月3日(星期天) 晚上6:30
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [AlessioDP/libby] Allow full-offline startup after resolved libraries once  (Issue #41)





 
Are you using the transitive dependency resolution feature?

**Comment by Ghost-chu** [comment]
> Are you using the transitive dependency resolution feature?

Yes, is this due to the need to check if its associated dependencies have all finished downloading?  
Can this be cached? 

As well as I noticed that libby always uses Maven Central when resolving its own dependencies, instead of the mirror specified by Dev, which defeats the purpose of mirroring and ends up causing huge delays for the user (the quality of our country/region's internet connection to Maven Central is pretty bad)...  
I'm using my own Fork to circumvent this problem.


Translated with DeepL.com (free version)

**Comment by frengor** [comment]
> As well as I noticed that libby always uses Maven Central when resolving its own dependencies, instead of the mirror specified by Dev

If you add your mirror before Maven Central the mirror should be queried before it.

**Comment by Ghost-chu** [comment]
> > As well as I noticed that libby always uses Maven Central when resolving its own dependencies, instead of the mirror specified by Dev
> 
> If you add your mirror before Maven Central the mirror should be queried before it.

It's load libraries into an standalone classloader and never use user defined mirrors.
